src/Tasks/Microsoft.NET.Build.Tasks/CompileReadyToRunImages.cs

```diff
@@ -0,0 +1,376 @@
+                $"/out \"{outputR2RImage}\" " +
+                $"\"{file.ItemSpec}\"";
+
+            return ExecuteTool(_crossgenPath, null, arguments) == 0;
+        }
+
+        private bool CreatePDBForImage(ITaskItem file, out string outputPDBImage)
+        {
+            outputPDBImage = null;
+
+            string arguments = null;
+
+            if (_targetPlatform == OSPlatform.Windows)
+            {
+                if (_diasymreaderPath == null || !File.Exists(_diasymreaderPath))
+                    return false;
+
+                outputPDBImage = Path.ChangeExtension(file.ItemSpec, "ni.pdb");
+
+                arguments = $"/nologo " +
+                    $"/Platform_Assemblies_Paths \"{_platformAssembliesPath}{_pathSeparatorCharacter}{Path.GetDirectoryName(file.ItemSpec)}\" " +
+                    $"/DiasymreaderPath \"{_diasymreaderPath}\" " +
+                    $"/CreatePDB \"{Path.GetDirectoryName(file.ItemSpec)}\" " +
+                    $"\"{file.ItemSpec}\"";
+            }
+            else if(_targetPlatform == OSPlatform.Linux)
+            {
+                using (FileStream fs = new FileStream(file.ItemSpec, FileMode.Open, FileAccess.Read))
+                {
+                    PEReader pereader = new PEReader(fs);
+                    MetadataReader mdReader = pereader.GetMetadataReader();
+                    Guid mvid = mdReader.GetGuid(mdReader.GetModuleDefinition().Mvid);
+
+                    outputPDBImage = Path.ChangeExtension(file.ItemSpec, "ni.{" + mvid + "}.map");
+                }
+
+                arguments = $"/nologo " +
+                    $"/Platform_Assemblies_Paths \"{_platformAssembliesPath}{_pathSeparatorCharacter}{Path.GetDirectoryName(file.ItemSpec)}\" " +
+                    $"/CreatePerfMap \"{Path.GetDirectoryName(file.ItemSpec)}\" " +
+                    $"\"{file.ItemSpec}\"";
+            }
+
+            return arguments == null ? false : (ExecuteTool(_crossgenPath, null, arguments) == 0);
+        }
+
+        bool ExtractTargetPlatformAndArchitectureFromRuntimeIdentifier()
+        {
+            int separator = _runtimeIdentifier.LastIndexOf('-');
+            if (separator < 0 || separator >= _runtimeIdentifier.Length)
+            {
+                return false;
+            }
+
+            string targetPlatform = _runtimeIdentifier.Substring(0, separator).ToLowerInvariant();
+            string targetArchitecture = _runtimeIdentifier.Substring(separator + 1).ToLowerInvariant();
+
+            if (targetPlatform.Contains("linux"))
+                _targetPlatform = OSPlatform.Linux;
+            else if (targetPlatform.Contains("osx"))
+                _targetPlatform = OSPlatform.OSX;
+            else if (targetPlatform.Contains("win"))
+                _targetPlatform = OSPlatform.Windows;
+            else
+                return false;
+
```

Comment:
Is rhel.6 intentionally unsupported? 

Do I have to be on a a linux-musl machine to run crossgen from linux-musl runtime pack? Does rhel have the same issue?

I think it would be better to switch over the runtime packs that are actually supported than to use Contains.

Reply:
rhel should be supported. This is just a bug here.

> Do I have to be on a a linux-musl machine to run crossgen from linux-musl runtime pack? Does rhel have the same issue?

I don't believe there's any linux flavor specific code in crossgen

> I think it would be better to switch over the runtime packs that are actually supported than to use Contains.
Agreed

Reply:
I rewrote this as we discussed, comparing the platform obtained from the runtime pack RID to the platform obtained from the host OS.